### DISCUSSION TITLE
Unsynchronized usual links structure and hierarchical links stucture

### DISCUSSION THREAD

**Discussion by Noone746458** [SkepticMystic/breadcrumbs#148] [Ideas]
*my english is not so really good, so there may be some mistakes or style of presentation may seem strange

That's great plugin and i glad that i found that. But there is some little problem. I've already have tag structure through crosslinks, using files (graph nodes) as tags. And everything i rly wanted is to have hierarchy between them. And this plugin gave me that. But i have to duplicate these tags that i've already have in another diffirent syntax. And that most unconvinient, if i need to edit them than i need to edit both of them. And i can't use default graph to new hierarchy structure but i still use default graph because sometimes its more convinient. So, i have 2 different unsynchronized structures and each of them has its own features, so i'm using both of them and in the same time i need to sync them to keep them up to date. Is it possible to somehow merge them? And it means to maybe add the ability to see hierarchy in default graph and/or change hierarchy syntax or somehowe insert kinda link of massive of usual links or it would be enough if hierarchical links also could be determined as usual links.

**Comment by SkepticMystic** [comment]
Hey Noone! Thanks for the message.
Can you please share an example of how you structure your notes, so I can get an idea of how to keep it in sync?

**Reply by Noone746458** [reply]
I have pool of notes and using special kind of nodes as tags. For now it kinda looks like

```
---
par: [[00 Tag1], [00 Tag2], [00 Tag3], [00 Tag4], [00 Tag5], [00 Tag6], [00 Tag7], [00 Tag8]]
---

### Tags
[[00 Tag1]] [[00 Tag2]] [[00 Tag3]] [[00 Tag4]] [[00 Tag5]] [[00 Tag6]] [[00 Tag7]] [[00 Tag8]]
```

But i need just node tags with hierarchy


I made an example
![1](https://user-images.githubusercontent.com/93587884/145973799-e4aefcc4-d433-4ce0-9c46-d5b30971ff46.png)
![2](https://user-images.githubusercontent.com/93587884/145973825-9e0f2e6b-3a61-434a-a573-8cd9cf227fc4.png)
![1m](https://user-images.githubusercontent.com/93587884/145973835-b839920c-a878-49ba-88f2-bffafedd5acb.png)




Main idea is to be able to visually display the hierarchy and/or highlight some of its individual elements
(e.g. all subnodes)
![3](https://user-images.githubusercontent.com/93587884/145973923-45d1d516-df1a-4fe5-aece-9d99a95feba0.png)
![3m](https://user-images.githubusercontent.com/93587884/145973929-b865511e-2bea-4bb4-8ade-5a398edbdf94.png)

but not like this
![4](https://user-images.githubusercontent.com/93587884/145974013-f1648c0a-77f2-474b-94e4-240827c26a07.png)
![4m](https://user-images.githubusercontent.com/93587884/145974015-4172ae01-2a3f-49ad-9d9f-df47b0231113.png)



(or e.g. all uppernodes)
![5](https://user-images.githubusercontent.com/93587884/145974160-dba0bce7-bd1f-48fa-bd8c-811476621bc1.png)
![5m](https://user-images.githubusercontent.com/93587884/145974169-a2246201-268d-4757-9efa-a083b49b759c.png)

In the examples, i just highlighted each specific node. But it would be great if it worked the same way as it does with a common file structure


So main idea is to be able to visually display the hierarchy and/or highlight some of its individual elements (e.g. all subnodes or all uppernodes) and also work with notes using this hierarchy (e.g. use search or make changes in all subnodes or all uppernodes)
Its kinda the same as they have a path like a normal linear tree structure and you can work with them the same way. But however this is not a linear hierarchy, and at the same time, nodes can have crosslinks regardless of the hierarchy


I have attached an example vault for more details

[Example.zip](https://github.com/SkepticMystic/breadcrumbs/files/7710556/Example.zip)

**Comment by michaelpporter** [comment]
v4 has builders for exactly this — reusing your existing link structure instead of duplicating it in a separate hierarchy syntax:

- **Traverse Note** (`BC-traverse-note-field: down` in a note's frontmatter): Breadcrumbs does a depth-first walk of the regular `[[links]]` reachable from that note and turns them into edges. Your existing crosslink structure becomes the hierarchy, no duplication.
- **List Note** (`BC-list-note-field: down`): the note's bullet list of links becomes the hierarchy.
- **Dataview Note** (`BC-dataview-note-query: outgoing([[#]])`): derive edges from a query over existing links.

So you can keep your single link structure and layer the hierarchy on top of it. Closing as resolved — see the docs for the Traverse / List / Dataview note builders.
